### PR TITLE
Fix memcheck error in TypeInference.Timestamp gtest

### PR DESCRIPTION
**Pull request by davidwendt**
## Description

Fixes an error in the `TypeInference.Timestamp` gtest where the `size` parameter was incorrect.

This error was found by the nightly builds and could be recreated using
```
compute-sanitizer --tool memcheck gtests/TYPE_INFERENCE_TEST --gtest_filter=TypeInference.Timestamp --rmm_mode=cuda
```

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/io/type_inference_test.cu [hunk 1]
```diff
@@ -211,7 +211,7 @@ TEST_F(TypeInference, Timestamp)
   auto d_data           = cudf::make_string_scalar(data);
   auto& d_string_scalar = static_cast<cudf::string_scalar&>(*d_data);
 
-  std::size_t constexpr size = 3;
+  std::size_t constexpr size = 2;
```
Comment (davidwendt) [inline cpp/tests/io/type_inference_test.cu:214]: Should this be set from the `string_offset` or `string_length` vector `size()` instead?
It appears they should be correlated but I'm not sure.

Comment (vuule) [reply cpp/tests/io/type_inference_test.cu:214]: Yeah, looks like this variable is not needed, using `string_offset.size()` in `infer_data_type` calls would be less error-prone.

Comment (davidwendt) [reply cpp/tests/io/type_inference_test.cu:214]: Ok. I think that is worth doing here.

**Review by vuule — APPROVED** [review]
Thank you!

**Review by ttnghia — APPROVED** [review]

**Pull request merged**